src/Core/Services/Implementations/StripePaymentService.cs

```diff
@@ -1429,6 +1430,22 @@ public async Task ReinstateSubscriptionAsync(ISubscriber subscriber)
             return createdCustomer;
         }
 
+        public async Task<bool> UpdateCustomerEmailAddress(string gatewayCustomerId, string emailAddress)
+        {
+            try
+            {
+                var customer = await _stripeAdapter.CustomerGetAsync(gatewayCustomerId);
+
+                await _stripeAdapter.CustomerUpdateAsync(customer.Id,
+                    new Stripe.CustomerUpdateOptions { Email = emailAddress });
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
```

Comment:
I wonder if we should move this to another service, since the `StripePaymentService` is already way to big and complex.

Reply:
I totally agree